src/main/java/org/jenkinsci/plugins/badge/PublicBadgeAction.java

```diff
@@ -92,61 +92,38 @@ public String getDisplayName() {
     /**
      * Serves the codeCoverage badge image.
      */
-    public HttpResponse doCoverageIcon(StaplerRequest req, StaplerResponse rsp, @QueryParameter String job, @QueryParameter String build, @QueryParameter String style) {
-    	if(build != null) {
-            Run run = getRun(job, build);
-            Integer codeCoverage = null;
-            
-            // Checks for Cobertura
-            CoberturaBuildAction coverageAction = run.getAction(CoberturaBuildAction.class);
-            if(coverageAction != null){
-              codeCoverage = new Integer(coverageAction.getBuildHealth().getScore());
-            }
-            // Checks for Clover
-            CloverBuildAction cloverAction = run.getAction(CloverBuildAction.class);
-            if (cloverAction != null){
-              codeCoverage = new Integer(cloverAction.getBuildHealth().getScore());
-            }
-            // Checks for Jacoco
-            JacocoBuildAction jacocoAction = run.getAction(JacocoBuildAction.class);
-            if (jacocoAction != null) {
-              codeCoverage = new Integer(jacocoAction.getInstructionCoverage().getPercentage());
-            }
-            return iconResolver.getCoverageImage(codeCoverage);
-        } else {
-            Job<?, ?> project = getProject(job);
-            Integer codeCoverage = null;
-            
-            // Checks for Cobertura
-            CoberturaBuildAction coverageAction = project.getLastSuccessfulBuild().getAction(CoberturaBuildAction.class);
-            if(coverageAction != null){
-              codeCoverage = new Integer(coverageAction.getBuildHealth().getScore());
-            }
-            // Checks for Clover
-            CloverBuildAction cloverAction = project.getLastSuccessfulBuild().getAction(CloverBuildAction.class);
-            if (cloverAction != null){
-              codeCoverage = new Integer(cloverAction.getBuildHealth().getScore());
-            }
-            // Checks for Jacoco
-            JacocoBuildAction jacocoAction = project.getLastSuccessfulBuild().getAction(JacocoBuildAction.class);
-            if (jacocoAction != null) {
-              codeCoverage = new Integer(jacocoAction.getInstructionCoverage().getPercentage());
-            }
-            
-            return iconResolver.getCoverageImage(codeCoverage);
+    public HttpResponse doCoverageIcon(StaplerRequest req, StaplerResponse rsp, @QueryParameter String job, @QueryParameter String style) {
+    	Job<?, ?> project = getProject(job);
+        Integer codeCoverage = null;
+        
+        // Checks for Cobertura
+        if (project.getLastSuccessfulBuild() != null) {
+	        CoberturaBuildAction coverageAction = project.getLastSuccessfulBuild().getAction(CoberturaBuildAction.class);
+	        if(coverageAction != null){
+	          codeCoverage = new Integer(coverageAction.getBuildHealth().getScore());
+	        }
```

Comment:
[Checkstyle] WARNING: 'if rcurly' have incorrect indentation level 16, expected level should be 6.